edi_oca/models/edi_exchange_record.py

```diff
@@ -603,3 +612,28 @@ def delayable(self, **kw):
 
     def _job_retry_params(self):
         return {}
+
+    def _compute_is_obsolete(self):
+        for rec in self:
+            rec.is_obsolete = (
+                rec.type_id.deduplicate_on_send
+                and rec.has_fresher_duplicates()
+                and not rec.block_obsolescence
+            )
+
+    def get_fresher_duplicates(self, count=None):
+        self.ensure_one()
+        fresher_duplicates = self.search(
+            [
+                ("id", ">", self.id),
+                ("res_id", "=", self.res_id),
+                ("model", "=", self.model),
+                ("type_id", "=", self.type_id.id),
+                ("exchange_file", "!=", False),
```

Comment:
I'm not sure about this check on the exchange_file. 1st it might be slow and 2nd... if the file is not there yet it means that it will be generated soon and it will be the newest record. No?